src/lib/comms/include/sol-connman.h

```diff
@@ -0,0 +1,285 @@
+ * Since the connman function is asynchronous, the return is not connection error,
+ * but just some dispatching/immediate error. The actual state change will be notified
+ * via sol_connman_add_service_monitor() callbacks. This callback must be added BEFORE
+ * the sol_connman_service_connect() is called to ensure no messages are lost.
+ *
+ * @param service The service structure which the connection is desired
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int sol_connman_service_connect(struct sol_connman_service *service);
+
+/**
+ * @brief Disconnect the service
+ *
+ * This function disconnects the service.
+ * Since the connman function is asynchronous, the return is not disconnection error,
+ * but just some dispatching/immediate error. The actual state change will be notified
+ * via sol_connman_add_service_monitor() callbacks. This callback must be added BEFORE
+ * the sol_connman_service_disconnect() is called to ensure no messages are lost.
+ *
+ * @param service The service structure which the disconnection is desired
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int sol_connman_service_disconnect(struct sol_connman_service *service);
+
+/**
+ * @brief Adds a monitor for the updated connman services
+ *
+ * To monitor the property of services, it gives information
+ * about the services. The callback is used to provide the information.
+ * sol_connman_add_service_monitor will call sol_connman_lazy_init
+ * to init connection manager. This callback must be added BEFORE
+ * the all functions are called to ensure connection manager
+ * has been initialized.
+ *
+ * @param cb monitor Callback to be called when the services are updated
+ * @param data Add a user data per callback
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int sol_connman_add_service_monitor(void (*cb)(void *data, const struct sol_connman_service *service),
+    const void *data);
+
+/**
+ * @brief Dels a monitor for the updated connman services
+ *
+ * Removes the monitor for updated connman services.
+ * sol_connman_del_service_monitor will call sol_connman_lazy_shutdown
+ * to shutdown connection manager. This callback must be added AFTER
+ * the all functions are called to ensure connection manager has been shutdown.
+ *
+ * @param cb monitor Callback to be called when the services are updated
+ * @param data Add a user data per callback
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int sol_connman_del_service_monitor(void (*cb)(void *data, const struct sol_connman_service *service),
+    const void *data);
+
+/**
+ * @brief Gets the connman services vector.
+ *
+ * The connman services vector can be given via sol_connman_get_vector.
+ *
+ * @param vector The retrieved content
+ *
+ * @return 0 if the content was successfully retrieved, error code (always negative) otherwise.
+ */
+int sol_connman_get_service_vector(struct sol_vector **vector);
```

Comment:
make this signature similar to sol-network.h `sol_network_get_available_links()`, see https://github.com/solettaproject/soletta/blob/master/src/lib/comms/include/sol-network.h#L273

Also say the vector is the last-known and more may be added/removed dynamically, then the pattern is to add a service monitor BEFORE calling this function.